File: readability/readability.py
from cs50 import get_string

text = get_string ("text: ")
lettre =0
word=0
sent =0
for c in text:
    if c.isalpha():
        lettre+=1
    elif c == " ":
        word+=1
    elif c in [".","!","?"]:
        sent+=1
word+=1;
l= lettre / word * 100
s = sent / word * 100
index = 0.0588 * l - 0.296 * s - 15.8

if index < 0:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print("Grade ",round(index))

File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>
int liau(string text);
int main (void){

    string text = get_string ("text: ");
    int x = liau(text);
    if (x < 0)
    printf("Before Grade 1\n");
    else if (x>16)
    printf("Grade 16+\n");
    else
    printf("Grade %i\n", x);
}

int liau(string text)
{
    int word=0;
    int sent = 0;
    int lettre=0;
  for (int i=0 , n=strlen(text) ; i < n ; i++)
    {
        if (isalpha(text[i]))
        lettre++;
        else if (text[i]==' ')
        word++;
        else if (text[i]=='.' || text[i]=='!' || text[i]=='?')
        sent++;

        }
        word++;
        float L = (float) lettre / (float) word * 100;
        float S = (float) sent / (float) word * 100;
        float index = 0.0588 * L - 0.296 * S - 15.8;
        return round(index);
}

